File: automator/src/main/java/com/stardust/automator/filter/ClassNameFilter.java
package com.stardust.automator.filter;

import com.stardust.automator.UiObject;

/**
 * Created by Stardust on 2017/3/9.
 */

public class ClassNameFilter {

    private static final KeyGetter CLASS_NAME_GETTER = new KeyGetter() {
        @Override
        public String getKey(UiObject nodeInfo) {
            CharSequence charSequence = nodeInfo.getClassName();
            return charSequence == null ? null : charSequence.toString();
        }

        @Override
        public String toString() {
            return "className";
        }
    };

    public static ListFilter equals(String text) {
        if (!text.contains(".")) {
            text = "android.widget." + text;
        }
        return new StringEqualsFilter(text, CLASS_NAME_GETTER);
    }

    public static ListFilter contains(String str) {
        return new StringContainsFilter(str, CLASS_NAME_GETTER);
    }

    public static ListFilter startsWith(String prefix) {
        return new StringStartsWithFilter(prefix, CLASS_NAME_GETTER);
    }

    public static ListFilter endsWith(String suffix) {
        return new StringEndsWithFilter(suffix, CLASS_NAME_GETTER);
    }

    public static ListFilter matches(String regex) {
        return new StringMatchesFilter(regex, CLASS_NAME_GETTER);
    }

    private ClassNameFilter() {

    }
}

File: automator/src/main/java/com/stardust/automator/filter/ClassNameFilter.kt
package com.stardust.automator.filter

import com.stardust.automator.UiObject

/**
 * Created by Stardust on 2017/3/9.
 */

object ClassNameFilter {

    private val CLASS_NAME_GETTER = object : KeyGetter {
        override fun getKey(nodeInfo: UiObject): String? {
            val charSequence = nodeInfo.className
            return charSequence?.toString()
        }

        override fun toString(): String {
            return "className"
        }
    }

    fun equals(text: String): ListFilter {
        var text = text
        if (!text.contains(".")) {
            text = "android.widget.$text"
        }
        return StringEqualsFilter(text, CLASS_NAME_GETTER)
    }

    fun contains(str: String): ListFilter {
        return StringContainsFilter(str, CLASS_NAME_GETTER)
    }

    fun startsWith(prefix: String): ListFilter {
        return StringStartsWithFilter(prefix, CLASS_NAME_GETTER)
    }

    fun endsWith(suffix: String): ListFilter {
        return StringEndsWithFilter(suffix, CLASS_NAME_GETTER)
    }

    fun matches(regex: String): ListFilter {
        return StringMatchesFilter(regex, CLASS_NAME_GETTER)
    }
}
